kartothek/serialization/_parquet.py

```diff
@@ -40,62 +37,28 @@
 def _empty_table_from_schema(parquet_file):
     schema = parquet_file.schema.to_arrow_schema()
 
-    if ARROW_LARGER_EQ_0150:
-        # ARROW-6872: empty dictionary columns raise ArrowNotImplementedError
-        for i in range(len(schema)):
-            field = schema[i]
-            if pa.types.is_dictionary(field.type):
-                new_field = pa.field(
-                    field.name, field.type.value_type, field.nullable, field.metadata
-                )
-                schema = schema.remove(i).insert(i, new_field)
-
     return schema.empty_table()
 
 
 def _reset_dictionary_columns(table, exclude=None):
     """
-    1. Categorical columns of null won't cast https://issues.apache.org/jira/browse/ARROW-5085
-    2. Massive performance issue due to non-fast path implementation https://issues.apache.org/jira/browse/ARROW-5089
-    3. We need to ensure that the dtype is exactly as requested, see GH227
+    We need to ensure that the dtype is exactly as requested, see GH227
     """
     if exclude is None:
         exclude = []
 
-    if ARROW_LARGER_EQ_0150:
-        # https://issues.apache.org/jira/browse/ARROW-8142
-        if len(table) == 0:
-            df = table.to_pandas(date_as_object=True)
-            new_types = {
-                col: df[col].cat.categories.dtype
-                for col in df.select_dtypes("category")
-            }
-            if new_types:
-                df = df.astype(new_types)
-                table = pa.Table.from_pandas(df)
-        else:
-            schema = table.schema
-            for i in range(len(schema)):
-                field = schema[i]
-                if pa.types.is_dictionary(field.type):
-                    new_field = pa.field(
-                        field.name,
-                        field.type.value_type,
-                        field.nullable,
-                        field.metadata,
-                    )
-                    schema = schema.remove(i).insert(i, new_field)
+    schema = table.schema
+    for i in range(len(schema)):
+        field = schema[i]
+        if field.name in exclude:
```

Comment:
In the previous code, it seems we ignored the exclude for arrow>=0.15.0.

Reply:
behaviour was the same but we worked more than necessary by converting it twice in this case. Added a test case since this has potentially huge performance implications for readers